src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -255,7 +255,7 @@ SnapTrimObjSubEvent::adjust_snaps(
   obc->obs.oi.prior_version = obc->obs.oi.version;
   obc->obs.oi.version = osd_op_p.at_version;
   ceph::bufferlist bl;
-  encode(obc->obs.oi,
+  obc->obs.oi.encode_no_oid(
```

Comment:
Shouldn't we `setattr` as `std::map<std::string, ceph::bufferlist, std::less<>> attrs;` here and not simply as a bufferlist?
```
  std::map<std::string, ceph::bufferlist, std::less<>> attrs;
  obc->obs.oi.encode_no_oid(..)
  attrs[OI_ATTR] = std::move(bl);
  txn.setattrs(
    pg->get_collection_ref()->get_cid(),
    ghobject_t{head_oid, ghobject_t::NO_GEN, shard_id_t::NO_SHARD},
    attrs);
```

Reply:
> Can you please clarify this bit? Is this related to SnapTrimObjSubEvent not mutating the obc?

Not really, this is actually a bug that has been covered up by the previous universal `mutate_object` process. Ever since 3aeea9b7e6f4, all `OI_ATTR` encoding should be without oid.

> If not, and we only need the encoding of `OI_ATTR` regardless, can you please move this into a separate commit?

Yeah, will do.

Reply:
> Unrelated to this PR, Shouldn't we `setattr` as `std::map<std::string, ceph::bufferlist, std::less<>> attrs;` here and not simply as a bufferlist?
> 
> ```
>   std::map<std::string, ceph::bufferlist, std::less<>> attrs;
>   obc->obs.oi.encode_no_oid(..)
>   attrs[OI_ATTR] = std::move(bl);
>   txn.setattrs(
>     pg->get_collection_ref()->get_cid(),
>     ghobject_t{head_oid, ghobject_t::NO_GEN, shard_id_t::NO_SHARD},
>     attrs);
> ```

Sorry, I don't quite follow, what is the purpose of this change?

Reply:
> Sorry, I don't quite follow, what is the purpose of this change?

My bad, I didn't notice we use `setattr` here and not `setattrs`. Please Ignore :)